src/main/java/com/icthh/xm/tmf/ms/communication/config/CommunicationTenantConfigService.java

```diff
@@ -0,0 +1,82 @@
+package com.icthh.xm.tmf.ms.communication.config;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.icthh.xm.commons.config.client.config.XmConfigProperties;
+import com.icthh.xm.commons.config.client.service.TenantConfigService;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.Data;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class CommunicationTenantConfigService extends TenantConfigService {
+
+    private final Map<String, CommunicationTenantConfig> configs = new ConcurrentHashMap<>();
+    private final TenantContextHolder tenantContextHolder;
+
+    public CommunicationTenantConfigService(XmConfigProperties xmConfigProperties,
+                                            TenantContextHolder tenantContextHolder) {
+        super(xmConfigProperties, tenantContextHolder);
+        this.tenantContextHolder = tenantContextHolder;
+    }
+
+    @Override
+    @SneakyThrows
+    public void onRefresh(String updatedKey, String config) {
+        try {
+            super.onRefresh(updatedKey, config);
+            ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
```

Comment:
Should it create every time on refresh?maybe better to create bean.

Reply:
moved objectMapper to a final class field and init it in constructor